be/src/exec/pipeline/pipeline_driver_queue.cpp

```diff
@@ -243,6 +250,9 @@ StatusOr<DriverRawPtr> WorkGroupDriverQueue::take() {
         _update_bandwidth_control_period();
 
         if (_wg_entities.empty()) {
+            if (!block) {
+                return nullptr;
+            }
```

Comment:
In the `else if` block, should also check `block` before `_cv.wait_for(lock, std::chrono::nanoseconds(sleep_ns));`?

Reply:
Yes, indeeed.